nodeJS/express/deployment.md

```diff
@@ -0,0 +1,300 @@
+### Introduction
+
+Before continuing our web development journey, we must address one important task - showcasing our hard work to the world.
+
+Whether it's to share our creations with friends, create a portfolio for future employers, or launch an online business, we need to host our applications somewhere others can publicly access on the internet.
+
+In this lesson, we will learn how to deploy our apps to a hosting provider, allowing us to run, build, and operate our web applications in the cloud.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Understand what a hosting provider is.
+- Understand the difference between static and dynamic sites.
+- Understand what a PaaS is and how they work.
+- Know how to deploy to a PaaS provider.
+- Know how to troubleshoot common deployment issues.
+
+### What are hosting providers?
+
+Hosting providers are like server landlords. They own servers and rent space on them to customers, who can then use the space to store their websites and make them accessible to anyone on the web.
+
+You've already had some experience using a hosting provider when you deployed projects to Github pages earlier in the curriculum. GitHub pages is great for hosting **static** web pages for free, but we won't be able to use it for hosting our **dynamic** NodeJS apps. We're going to need something more powerful.
+
+### Static vs dynamic sites
+
+Static websites consist of pre-written HTML pages. They are "static" because everyone who visits them will see the same content. To build static sites, you only need HTML, CSS and JavaScript.
+
+Dynamic websites, on the other hand, are websites that can change content based on the user who is visiting them. X (formerly known as Twitter) is a good example; every user sees different content on their homepage feed based on who they follow. To build dynamic sites, you still need HTML, CSS and JS. But additionally, you need a server-side application and a database.
+
+This additional tech prohibits us from using GitHub Pages for hosting our NodeJS apps. GitHub pages cannot run NodeJS applications and doesn't have database services we can use.
```

Comment:
One last thing from me, just adding this in to extend the PaaS justification to cover "why not use Netlify/Vercel?" which is a somewhat common question for new starters.

```suggestion
This additional tech prohibits us from using GitHub Pages for hosting our NodeJS apps. GitHub pages cannot run Node.js applications and doesn't have database services we can use. Similarly, host providers like Netlify and Vercel, which you may have used back in the React course, do not have the same capabilities to run our Node.js servers and databases. They're not the right tools for our back ends.
```